server/app/views/admin/programs/ProgramTranslationView.java

```diff
@@ -102,55 +101,65 @@ private ImmutableList<DomContent> formFields(
                         div()
                             .with(
                                 FieldWithLabel.input()
-                                    .setFieldName("displayName")
+                                    .setFieldName(ProgramTranslationForm.DISPLAY_NAME_FORM_NAME)
                                     .setLabelText("Program name")
-                                    .setValue(translationForm.getDisplayName())
+                                    .setValue(updateData.localizedDisplayName())
                                     .getInputTag(),
                                 defaultLocaleTextHint(program.localizedName())),
                         div()
                             .with(
                                 FieldWithLabel.input()
-                                    .setFieldName("displayDescription")
+                                    .setFieldName(
+                                        ProgramTranslationForm.DISPLAY_DESCRIPTION_FORM_NAME)
                                     .setLabelText("Program description")
-                                    .setValue(translationForm.getDisplayDescription())
+                                    .setValue(updateData.localizedDisplayDescription())
                                     .getInputTag(),
                                 defaultLocaleTextHint(program.localizedDescription())))));
     if (statusTrackingEnabled.get()) {
       String programStatusesLink =
           controllers.admin.routes.AdminProgramStatusesController.index(program.id()).url();
 
-      for (StatusDefinitions.Status s : program.statusDefinitions().getStatuses()) {
+      Preconditions.checkState(
+          updateData.statuses().size() == program.statusDefinitions().getStatuses().size());
+      for (int statusIdx = 0; statusIdx < updateData.statuses().size(); statusIdx++) {
+        StatusDefinitions.Status configuredStatus =
+            program.statusDefinitions().getStatuses().get(statusIdx);
+        LocalizationUpdate.StatusUpdate statusUpdateData = updateData.statuses().get(statusIdx);
         ImmutableList.Builder<DomContent> fieldsBuilder =
             ImmutableList.<DomContent>builder()
                 .add(
+                    input()
+                        .isHidden()
```

Comment:
How does the hidden input associated with the visible input?

Reply:
Added a comment. Also renamed the method to `statusKeyToUpdateFieldName` per earlier feedback.

Reply:
Ah, yes. They're all siblings in the form, the main thing grouping them together is the index. So there will be the following inputs (much DOM ommitted for clarity):

````
<form>
  <input name="status-key-to-update-0" value="first-status" hidden />
  <input name="localized-status-0" value="localized-first-status" />
  <input name="localized-email-0" value="localized-first-status-email-body" />
  
  <input name="status-key-to-update-1" value="second-status" hidden />
  <input name="localized-status-1" value="localized-second-status" />
  <input name="localized-email-1" value="localized-second-status-email-body" />
</form>
```

Reply:
Left a comment to try to clarify this.